src/data/hash_queue.cc

```diff
@@ -11,8 +11,10 @@
 #include "torrent/exceptions.h"
 #include "torrent/utils/log.h"
 
-#define LT_LOG_DATA(data, log_level, log_fmt, ...)                       \
-  lt_log_print_data(LOG_STORAGE_##log_level, data, "hash_queue", log_fmt, __VA_ARGS__);
+#define LT_LOG_DATA(data, log_level, log_fmt, ...)                                        \
+  do {                                                                                    \
+    lt_log_print_data(LOG_STORAGE_##log_level, data, "hash_queue", log_fmt, __VA_ARGS__); \
+  } while (false)
```

Comment:
Is this really necessary?

Reply:
Alternative is a static void function

Reply:
I removed these changes.

Reply:
Please don't add additional unrelated changes after I've reviewed a PR, this makes it way more complicated than if those changes were to be a separate PR.

Reply:
Huh? The changes are the same since posting. I only modified this commit to remove the do while loops

Reply:
It looked like there was new changes there, however it might just be the original PR was so big I missed them.

In any case, there are too many different kinds of changes in this PR to make it a good merge.